programs/zstdcli.c

```diff
@@ -130,8 +130,9 @@ static void checkLibVersion(void)
    */
 static int exeNameMatch(const char* exeName, const char* test)
 {
-    return !strncmp(exeName, test, strlen(test)) &&
-        (exeName[strlen(test)] == '\0' || exeName[strlen(test)] == '.');
+    size_t len = strlen(test);
+    return !strncmp(exeName, test, len) &&
+        (exeName[len] == '\0' || exeName[len] == '.');
```

Comment:
The compiler will take care of this optimization, so I don't think we need to change this code.